### PR TITLE
Fix order of bind mount arguments

### PR DESCRIPTION
**Pull request by anz-ableton**
The arguments were previously applied in the wrong order, i.e. `/nix` would have been mounted onto `nix_bind_mount` and not the other way around. The PR also adds a test case that verifies that the correct directory is mounted.

### CODE REVIEW

File: molecule/default/verify.yml [hunk 1]
```diff
@@ -116,6 +116,18 @@
...
+      ansible.builtin.slurp:
+        src: "/proc/mounts"
+      register: proc_mounts
+
+    - name: Verify overlay mount for /nix
+      ansible.builtin.assert:
+        fail_msg: "/proc/mounts did not contain an overlay for /nix"
+        that:
+          - "{{ proc_mounts['content'] | b64decode \
+            | regex_search('overlay /nix') is not none }}"
```
Comment (nre-ableton) [inline molecule/default/verify.yml:129]: Why is this not `None`?

**Review by nre-ableton — APPROVED** [review]
Very nice! Just one question.

File: molecule/default/verify.yml (hunk 1 above)
Comment (anz-ableton) [reply molecule/default/verify.yml:129]: This is what the [Ansible documentation](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_filters.html#searching-strings-with-regular-expressions) suggests.

Comment (anz-ableton) [reply molecule/default/verify.yml:129]: I tried comparing against `None` and this actually fails.

Comment (nre-ableton) [reply molecule/default/verify.yml:129]: Ok, thanks for checking!

**Pull request merged**